go/vt/vtqueryserver/plugin_mysql_server.go

```diff
@@ -71,16 +75,17 @@ func (mh *proxyHandler) NewConnection(c *mysql.Conn) {
 
 func (mh *proxyHandler) ConnectionClosed(c *mysql.Conn) {
 	// Rollback if there is an ongoing transaction. Ignore error.
-	ctx := context.Background()
+	ctx, cancel := context.WithTimeout(context.Background(), *mysqlQueryTimeout)
```

Comment:
i think we want cleanup to succeed at all costs, especially since this is already in a defer.

granted, it can always fail for other reasons which is why we have transactionKiller, etc, but in general this should be something that can take a while and is not expensive.

Reply:
that's the suggestion, yeah. although now that i think about it this includes getting from the pool, etc so we probably do want a timeout on that. feel free to ignore.